File: scripts/snp-density.py
#!/usr/bin/env python3

import pandas as pd

# Load the .bim PLINK data
bim_file = './glabra-only_LD-pruned.bim'
bim_data = pd.read_csv(bim_file, delim_whitespace=True, header=None,
                       names=['CHR', 'SNP_ID', 'GEN_DIST', 'BP_POS', 'A1', 'A2'])

# Prepare DataFrame as required by CMplot R package
cmplot_data = bim_data[['SNP_ID', 'CHR', 'BP_POS']]
cmplot_data.columns = ['name', 'chromosome', 'position']

# Save the data to a CSV file or use the DataFrame directly in R
cmplot_data.to_csv('glabra-only_LD-pruned.bim.csv', index=False)

File: scripts/snp-density.R
# This script reads the CSV data and creates a SNP density plot.
# Install and load CMplot package 
# https://github.com/YinLiLin/CMplot

install.packages("CMplot")

library(CMplot)

# Read the input data
data <- read.csv('glabra-only_LD-pruned.bim.csv', header = TRUE, sep = ",")

# Plot SNP density
CMplot(data, plot.type="d", bin.size=1e6, chr.den.col=c("darkgreen", "yellow", "red"),
       file="pdf", file.name="SNP_Density_glabra.pdf", dpi=300, main="SNP Density , U. glabra", file.output=TRUE,verbose=TRUE,width=9,height=3,legend.cex=0.6)


